### ISSUE TITLE
Possible package issue beeing built for unsupported platforms

### ISSUE THREAD

**Issue by hussfelt** [asku/cordova-intercom#3]
As soon as I add the pacakge (after solving #1) I get this error:

```
      -[ICMSSKeychainQuery query] in libIntercom.a(ICMSSKeychainQuery.o)
  "_kSecMatchLimit", referenced from:
      -[ICMSSKeychainQuery fetchAll:] in libIntercom.a(ICMSSKeychainQuery.o)
      -[ICMSSKeychainQuery fetch:] in libIntercom.a(ICMSSKeychainQuery.o)
  "_kSecMatchLimitAll", referenced from:
      -[ICMSSKeychainQuery fetchAll:] in libIntercom.a(ICMSSKeychainQuery.o)
  "_kSecMatchLimitOne", referenced from:
      -[ICMSSKeychainQuery fetch:] in libIntercom.a(ICMSSKeychainQuery.o)
  "_kSecReturnAttributes", referenced from:
      -[ICMSSKeychainQuery fetchAll:] in libIntercom.a(ICMSSKeychainQuery.o)
  "_kSecReturnData", referenced from:
      -[ICMSSKeychainQuery fetch:] in libIntercom.a(ICMSSKeychainQuery.o)
  "_kSecValueData", referenced from:
      -[ICMSSKeychainQuery save:] in libIntercom.a(ICMSSKeychainQuery.o)
ld: symbol(s) not found for architecture i386
clang: error: linker command failed with exit code 1 (use -v to see invocation)

** BUILD FAILED **
```

It seems related to app trying to build for platforms that does not exist?

This is one of all the related issues on the subject:
http://stackoverflow.com/questions/22334680/xcode-5-1-libcordova-a-architecture-problems

Neither of them helped me - any clues?

**Comment by hussfelt** [comment]
Same thing now that I am able to build everything else and push to app-store, not only running in emulator.

```
...
      -[ICMFlowLayout finalLayoutAttributesForDisappearingSupplementaryElementOfKind:atIndexPath:] in libIntercom.a(ICMFlowLayout.o)
  "_SecTrustCopyPublicKey", referenced from:
      -[ICM_AFSecurityPolicy setPinnedCertificates:] in libIntercom.a(ICM_AFSecurityPolicy.o)
      -[ICM_AFSecurityPolicy evaluateServerTrust:forDomain:] in libIntercom.a(ICM_AFSecurityPolicy.o)
  "_SecTrustSetAnchorCertificates", referenced from:
      -[ICM_AFSecurityPolicy evaluateServerTrust:forDomain:] in libIntercom.a(ICM_AFSecurityPolicy.o)
  "_SecPolicyCreateBasicX509", referenced from:
      -[ICM_AFSecurityPolicy setPinnedCertificates:] in libIntercom.a(ICM_AFSecurityPolicy.o)
      -[ICM_AFSecurityPolicy evaluateServerTrust:forDomain:] in libIntercom.a(ICM_AFSecurityPolicy.o)
  "_OBJC_CLASS_$_CAShapeLayer", referenced from:
      objc-class-ref in libIntercom.a(ICMChatHeadButton.o)
      objc-class-ref in libIntercom.a(ICMConversationBaseCell.o)
      objc-class-ref in libIntercom.a(ICMBadgeView.o)
      objc-class-ref in libIntercom.a(ICMSmallAnnouncementCell.o)
  "_kSecMatchLimit", referenced from:
      -[ICMSSKeychainQuery fetchAll:] in libIntercom.a(ICMSSKeychainQuery.o)
      -[ICMSSKeychainQuery fetch:] in libIntercom.a(ICMSSKeychainQuery.o)
  "_SecTrustCreateWithCertificates", referenced from:
      -[ICM_AFSecurityPolicy setPinnedCertificates:] in libIntercom.a(ICM_AFSecurityPolicy.o)
      -[ICM_AFSecurityPolicy evaluateServerTrust:forDomain:] in libIntercom.a(ICM_AFSecurityPolicy.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by elliotstokes** [comment]
I think there are some missing dependencies. Security.framework and QuartzCore.framework. After I had added them it worked fine.

**Comment by hussfelt** [comment]
@elliotstokes How did you add those?

**Comment by elliotstokes** [comment]
I did a cordova prepare and then added them manually via xcode. I think you can also add some references to the plugin.xml although I haven't tried it yet.

**Comment by hussfelt** [comment]
Ah.

We're looking to automate this, without any need to open Xcode...
I am currently having this feature on hold, so if you figure something neat out - please do share! :-) :+1:

**Comment by joshdover** [comment]
@elliotstokes Ah those just need to be added to the plugin.xml like the other frameworks. eg:

```
<framework src="Security.framework" />
```

Thanks for catching that.

**Comment by hussfelt** [comment]
@joshdover @elliotstokes I have not yet got around to switching from Cordova (from trigger), I'll try this again soon!

Created a PR anyway so that you wouldn't need to do the change:
https://github.com/asku/cordova-intercom/pull/9

Thanks guys!

**Comment by joshdover** [comment]
Fixed by #9
